### PR TITLE
Fix: Move device error message code to HardwareSetsControl

### PR DESCRIPTION
**Pull request by alexdewar**
# Description

Currently, the code responsible for showing an error message when a device error occurs lives in the DeviceControl class, where it always has. The problem is that nowadays a DeviceControl is only created when the user clicks "Manage devices", meaning that otherwise no error dialogs will be shown for device errors.

Fix by moving the relevant code to the HardwareSetsControl class instead.

## Type of change

- [ ] New feature (non-breaking change which adds functionality)
- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] Documentation (non-breaking change that adds or improves the documentation)
- [ ] Optimisation (non-breaking, back-end change that speeds up the code)

## Key checklist

- [x] Pre-commit hooks run successfully (`pre-commit run -a`)
- [x] All tests pass (`pytest`)
- [ ] The documentation builds without warnings (`mkdocs build -s`)
- [x] Check the GUI still works (if relevant)
- [x] Check the code works with actual hardware (if relevant)
- [ ] Check the `pyinstaller`-built executable works (if relevant)

## Further checks

- [ ] Code is commented, particularly in hard-to-understand areas
- [x] Tests have been added or an issue has been opened to tackle that in the future. (Indicate issue here: # (issue))

### CODE REVIEW

File: finesse/gui/hardware_set/hardware_sets_view.py [hunk 1]
```diff
@@ -75,6 +75,7 @@ def __init__(self) -> None:
         self._connected_devices: set[OpenDeviceArgs] = set()
         pub.subscribe(self._on_device_opened, "device.opening")
         pub.subscribe(self._on_device_closed, "device.closed")
+        pub.subscribe(self._on_device_error, "device.error")
```
Comment (dalonsoa) [inline finesse/gui/hardware_set/hardware_sets_view.py:78]: The original version used `f"device.error.{instance!s}"` as topic, and not just `f"device.error"`. Is this change on purpose?

**Review by dalonsoa — APPROVED** [review]
LGTM!!

File: finesse/gui/hardware_set/hardware_sets_view.py (hunk 1 above)
Comment (alexdewar) [reply finesse/gui/hardware_set/hardware_sets_view.py:78]: Yep. This handler will catch errors for *all* devices, as opposed to just the one specific type. `instance` is also passed as an argument, as well as part of the topic name, so we can distinguish devices that way.
